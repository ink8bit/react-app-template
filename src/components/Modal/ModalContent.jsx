import React from 'react';
import styled from '@emotion/styled';

import color from '../../styles/color';
import offset from '../../styles/offset';

const ModalContainer = styled.div`
  background: ${color.MODAL_BG};
  color: ${color.MODAL_TEXT};
  position: absolute;
  transform: translate(-50%, -50%);
  top: 50%;
  left: 50%;
  width: 300px;
  height: 200px;
  padding: ${offset.PADDING_COMMON};
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  border-radius: 4px;

  h1 {
    font-size: 20px;
  }

  .buttons {
    display: flex;
    align-self: flex-end;
  }

  .button {
    padding: 4px 8px;
    border-radius: 4px;
    flex-shrink: 0;
    cursor: pointer;

    &:hover {
      box-shadow: 1px 1px 1px 1px ${color.MODAL_BUTTON_SHADOW};
    }
  }

  .button + .button {
    margin-left: 10px;
  }
`;

const ModalContent = props => (
  <ModalContainer>
    <h1>This is an example of a modal window</h1>
    <div className="buttons">
      <button className="button" onClick={props.toggleModal}>
        Ok
      </button>
      <button className="button" onClick={props.toggleModal}>
        Close
      </button>
    </div>
  </ModalContainer>
);

export default ModalContent;
