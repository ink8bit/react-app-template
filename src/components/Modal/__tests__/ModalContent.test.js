import React from 'react';
import { create } from 'react-test-renderer';

import ModalContent from './../ModalContent.jsx';

test('snapshot', () => {
  const c = create(<ModalContent />);
  expect(c.toJSON()).toMatchSnapshot();
});
