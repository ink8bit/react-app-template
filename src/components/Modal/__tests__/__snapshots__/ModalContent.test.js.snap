// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="css-1rvx2q7"
>
  <h1>
    This is an example of a modal window
  </h1>
  <div
    className="buttons"
  >
    <button
      className="button"
    >
      Ok
    </button>
    <button
      className="button"
    >
      Close
    </button>
  </div>
</div>
`;
